sphinx/texinputs/Makefile_t

```diff
@@ -7,34 +7,35 @@ ALLPS  = $(addsuffix .ps,$(ALLDOCS))
 
 # Prefix for archive names
 ARCHIVEPRREFIX =
-# Additional LaTeX options
+# Additional LaTeX options.
 LATEXOPTS =
 # format: pdf or dvi
 FMT = pdf
 
-LATEX = latex
-PDFLATEX = {{ latex_engine }}
-MAKEINDEX = makeindex
+LATEX = latexmk -dvi
+PDFLATEXCMD = {{ latex_engine }} --halt-on-error --interaction=nonstopmode
+PDFLATEX = latexmk -pdflatex="$(PDFLATEXCMD) %O %S" -pdf -dvi- -ps-
 
 {% if latex_engine == 'platex' %}
 all: all-pdf-ja
 all-pdf: all-pdf-ja
+# latexmkrc is read then overridden by latexmkjarc
+LATEX = latexmk -r latexmkjarc -pdfdvi -dvi- -ps-
 {% else %}
 all: $(ALLPDF)
 all-pdf: $(ALLPDF)
 {% endif -%}
+
+{% if latex_engine == 'lualatex' %}
+LATEXOPTS += -lualatex
+{% endif -%}
+
 all-dvi: $(ALLDVI)
 all-ps: $(ALLPS)
 
 all-pdf-ja:
 	for f in *.pdf *.png *.gif *.jpg *.jpeg; do extractbb $$f; done
```

Comment:
This looks very good idea indeed. Memo: since TeXLive 2015 dvipdfmx does not need the extractbb runs. But it needs them for earlier releases.